google/cloud/storage/examples/storage_object_samples.cc

```diff
@@ -80,24 +80,27 @@ void GetObjectMetadata(gcs::Client client, int& argc, char* argv[]) {
   (std::move(client), bucket_name, object_name);
 }
 
-//! [read object]
 void ReadObject(gcs::Client client, int& argc, char* argv[]) {
   if (argc < 2) {
-    throw Usage{
-        "insert-object <bucket-name> <object-name> <object-contents (string)>"};
+    throw Usage{"read-object <bucket-name> <object-name>"};
   }
   auto bucket_name = ConsumeArg(argc, argv);
   auto object_name = ConsumeArg(argc, argv);
-  auto stream = client.Read(bucket_name, object_name);
-  int count = 0;
-  while (not stream.eof()) {
-    std::string line;
-    std::getline(stream, line, '\n');
-    ++count;
+  //! [read object] [START download_file]
```

Comment:
I'd be careful with this sample. It doesn't show you how to download the object into a file. Instead it show you how to access the downloaded object as a stream. What do you think about saving the stream to a file instead of counting the number of lines? Given not all objects will be readable this way.

Reply:
We are planning to have a separate API to download objects directly to a file, I would rather save that example for that API.

I am curious, what objects would not be readable this way?  You mean binaries that do not have lines in them?  That is true, but most C++ programmers would know (or should know) that iostreams can be used to read and write both binary and text files.

In any case, we can change this to:

```C++
std::ofstream output("foobar.txt");
...
while (not stream.eof()) {
  std::getline(stream, line, '\n');
  output << line << "\n";
}
```

or more efficient:

```C++
std::ofstream output("foobar.txt");
...
while (not stream.eof()) {
  char buffer[4096];
  stream.read(buffer, 4096);
  output.write(buffer, buffer.gcount()); 
}
```

But we are not showing how to process the media in the code, which I think it is an interesting case.  I mean, if you just want to download to a file, `gsutil` is easier than writing C++ :grin:

Reply:
Sgtm! Let's keep as-is and remove tags from this PR.